### PR TITLE
feat: claim expieration

### PR DESCRIPTION
**Pull request by Harasz**
### Description

Add claim and credentials expiration.

### Contributor checklist

- [x] Breaking changes - check for any existing interfaces changes that are not backward compatible, removed method etc.
- [x] Documentation - document your code, add comments for method, remember to check if auto generated docs were updated.
- [x] Tests - add new or updated existed test for changes you made.
- [x] Migration guide - add migration guide for every breaking change.
- [x] Configuration correctness - check that any configuration changes are correct ex. default URLs, chain ids, smart contract verification on [Volta explorer](https://volta-explorer.energyweb.org/) or [EWC explorer](https://explorer.energyweb.org/).

### CODE REVIEW

**Review by JGiter — APPROVED** [review]

File: src/modules/claims/claims.service.ts [hunk 1]
```diff
@@ -392,7 +393,7 @@ export class ClaimsService {
     const { claimType: role, claimTypeVersion: version } = claimData;
 
     if (registrationTypes.includes(RegistrationTypes.OnChain)) {
-      const expiry = defaultClaimExpiry;
+      const expiry = expirationTimestamp || defaultClaimExpiry;
```
[the remark this answers is not in the document]
Comment (Harasz) [reply src/modules/claims/claims.service.ts:396]: We use `expirationTimestamp` in other places, i left expiry time just for on-chain claims as it was before

File: src/modules/claims/claims.types.ts [hunk 2]
```diff
@@ -241,6 +247,12 @@ export interface IssueClaimOptions {
     /** Issuers fields that role is requiring */
     issuerFields?: { key: string; value: string | number }[];
   };
+
+  /** Indicates if credential is actual of the time of verification */
```
Comment (jrhender) [inline src/modules/claims/claims.types.ts:251]:
```suggestion
  /*
   * Indicates credential status (such as revocation status)
   * https://w3c-ccg.github.io/vc-status-list-2021/#statuslist2021entry
   */
```

File: e2e/claims.service.e2e.ts [hunk 3]
```diff
@@ -507,6 +544,23 @@ describe('Enrollment claim tests', () => {
       ).rejects.toEqual(new Error(ERROR_MESSAGES.ROLE_PREREQUISITES_NOT_MET));
     });
 
+    test.only('enrollment with credential/claim time expiration', async () => {
```
Comment (jrhender) [inline e2e/claims.service.e2e.ts:547]: Is the `.only` correct?
```suggestion
    test('enrolment with credential/claim time expiration', async () => {
```

**Review by jrhender — APPROVED** [review]
Some suggestions but it looks good overall

File: e2e/claims.service.e2e.ts (hunk 3 above)
Comment (Harasz) [reply e2e/claims.service.e2e.ts:547]: I removed it here https://github.com/energywebfoundation/iam-client-lib/pull/583

**Pull request merged**